src/main/java/org/haffson/adventofcode/days/day06/Coordinate.java

```diff
@@ -0,0 +1,358 @@
+package org.haffson.adventofcode.days.day06;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Helper record for day 6.
+ * Is used to work better with the string inputs.
+ * Coordinates have an x and y value.
+ * Also, you can compare two coordinates with each other.
+ */
+public record Coordinate(int xCoordinate, int yCoordinate) {
+
+    @Nonnull
+    public static Coordinate of(@Nonnull final String input) {
```

Comment:
Denk daran, auf Null zu checken. Das solltest du zumindest bei allen Public-Methoden jeder Klasse machen. Stichwort Fail-fast.

```suggestion
    public static Coordinate of(@Nonnull final String input) {
        requireNonNull(input, "input");
```

Reply:
Auch bei dir letztes Mal gesehen, aber noch nicht konsequent genutzt.
Diesen check mache ich nur, wenn die public Methode parameter hat und diese keinen primitiven Datentyp wie int etc, haben, richtig?

Reply:
Genau. Der grundsätzliche Gedanke ist: Offensichtlich fehlerhafte Aufrufe sollten so schnell wie möglich fehlschlagen.

Bei privaten Methoden kann man sich den Overhead in der Regel sparen, da man klassen-intern verinnerlicht haben sollte, wie die Methode korrekt aufgerufen wird. 
Ausnahme: Du arbeitest mit Daten, denen du nicht traust oder bei denen du nicht sicher bist, wie sie genau aussehen.

Wenn man direkt mit dem Parameter arbeitet, dann bekommt man auch sehr schnell einen Fehler:

     public static Coordinate of(@Nonnull final String input) {
        final var length = input.length(); // Schlägt sofort fehl, wenn input null ist
        // ...
     }

Um konsequent zu sein kann man hier den Null-Check hinzufügen. Dann muss man auch nicht immer groß nachdenken, ob Null-Check oder nicht.

Interessanter werden die Null-Checks bei Konstruktur-Aufrufen:
    
    public static Coordinate(@Nonnull final String input) {
        this.input = input; // Wenn input null ist merkt man den Fehler erst viel später, bei den Aufrufen anderer Methoden
    }

Ohne Check kann man hier viel Zeit verlieren, da man im Fehlerfall schnell an der falschen Stelle sucht (in der Methode, in der der Fehler dann zeitlich verzögert auftritt).

Reply:
Tatsächlich ist genau diese Problematik schon öfters vorgekommen. 
Methoden liefern mir nicht das gewünschte Ergebnis und nach langem debuggen landet man bei der obersten Methode und stellt fest, der FileReader hat einem null mitgegeben und der Fehler fällt erst nach x-Funktionen auf.
Werde ich mir definitiv angewöhnen früher zu handlen!